lib/wasi/src/state/mod.rs

```diff
@@ -44,6 +55,7 @@ pub struct InodeVal {
     pub kind: Kind,
 }
 
+/// The core file-object data type
```

Comment:
'core file-object' doesn't help me distinguish it from other "file" data types. Is it a file descriptor or some sort of handle? Or the part that holds the file's contents?